code/game/objects/items/devices/chameleonproj.dm

```diff
@@ -171,6 +171,81 @@
 	master.disrupt(0)
 	return ..()
 
+/obj/item/chameleon_counterfeiter
+	name = "chameleon counterfeiter"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "cham_counter"
+	flags = CONDUCT
+	slot_flags = SLOT_BELT
+	item_state = "electronic"
+	w_class = WEIGHT_CLASS_SMALL
+	origin_tech = "syndicate=3;magnets=3"
+	var/can_use = TRUE
+	var/saved_item
+	var/saved_icon
+	var/saved_icon_state
+	var/saved_item_state
+	var/saved_overlays
+	var/saved_underlays
+	var/activate_dummy = FALSE
+
+/obj/item/chameleon_counterfeiter/afterattack(obj/item/target, mob/user, proximity)
+	if(!proximity || !check_sprite(target) || target.alpha < 255 || target.invisibility != 0)
+		return
+	if(activate_dummy || !isitem(target))
+		return
+	playsound(get_turf(src), 'sound/weapons/flash.ogg', 100, 1, -6)
+	to_chat(user, "<span class='notice'>Scanned [target].</span>")
+	saved_item = target.type
+	saved_icon = target.icon
+	saved_icon_state = target.icon_state
+	saved_item_state = target.item_state
+	saved_overlays = target.overlays
+	saved_underlays = target.underlays
```

Comment:
Wouldn't saving `target.appearance` work as well here? Instead of all of the below

Reply:
Tested and it didn't work properly.